spec/rubocop/cop/rspec/capybara/specific_matcher_spec.rb

```diff
@@ -60,20 +60,72 @@
+     filter_set focused href href alt title download].each do |attr|
+    it 'registers an offense for abstract matcher when ' \
+      "first argument is link element with #{attr}" do
+      expect_offense(<<-RUBY, attr: attr)
+        expect(page).to have_css("a[#{attr}]")
+                        ^^^^^^^^^^^^^{attr}^^^ Prefer `have_link` over `have_css`.
+      RUBY
+    end
+  end
+
+  %i[above below left_of right_of near count minimum maximum between text id
+     class style visible obscured exact exact_text normalize_ws match wait
+     filter_set focused caption with_cols cols with_rows rows].each do |attr|
+    it 'registers an offense for abstract matcher when ' \
+      "first argument is table element with #{attr}" do
+      expect_offense(<<-RUBY, attr: attr)
+        expect(page).to have_css("table[#{attr}]")
+                        ^^^^^^^^^^^^^^^^^{attr}^^^ Prefer `have_table` over `have_css`.
+      RUBY
+    end
+  end
+
+  %i[above below left_of right_of near count minimum maximum between text id
+     class style visible obscured exact exact_text normalize_ws match wait
+     filter_set focused disabled name placeholder options enabled_options
+     disabled_options selected with_selected name placeholder disabled
+     multiple with_options].each do |attr|
+    it 'registers an offense for abstract matcher when ' \
+      "first argument is select element with #{attr}" do
+      expect_offense(<<-RUBY, attr: attr)
+        expect(page).to have_css("select[#{attr}]")
+                        ^^^^^^^^^^^^^^^^^^{attr}^^^ Prefer `have_select` over `have_css`.
+      RUBY
+    end
+  end
+
   it 'registers an offense when using abstract matcher with state' do
     expect_offense(<<-RUBY)
-    expect(page).to have_css('button[disabled]', exact_text: 'foo')
-                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer `have_button` over `have_css`.
-    expect(page).to have_css('button:not([disabled])', exact_text: 'bar')
-                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer `have_button` over `have_css`.
+      expect(page).to have_css('button[disabled]', exact_text: 'foo')
+                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer `have_button` over `have_css`.
+      expect(page).to have_css('button:not([disabled])', exact_text: 'bar')
+                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer `have_button` over `have_css`.
     RUBY
   end
 
   it 'does not register an offense for abstract matcher when ' \
     'first argument is element with nonreplaceable attributes' do
     expect_no_offenses(<<-RUBY)
-      expect(page).to have_css('button[foo=bar]')
-      expect(page).to have_css('button[foo-bar=baz]', exact_text: 'foo')
+    expect(page).to have_css('button[data_disabled]')
+    expect(page).to have_css('button[foo=bar]')
+    expect(page).to have_css('button[foo-bar=baz]', exact_text: 'foo')
```

Comment:
Cosmetic: I really prefer it indented as above.

Reply:
Oops, fixed.